### PR TITLE
[PVR][video] Consolidate implementation of "Play", "Mark watched", "Mark unwatched", "Resume from" context menu items

### PR DESCRIPTION
**Pull request by ksooo**
Both video files and pvr recordings have similar context menu items, like "Play", "Mark watched", "Mark unwatched" and "Resume from". For historical reasons, there are different implementations for these menu items, although lots of code could be shared.

After #10870 got merged it is now possible to consolidate these implementations.  This PR does exactly this. :-) A bunch of code refactoring needed as prerequisite for this is also included here.

A nice visual side effect of these changes is that now "Resume" and "Play" for PVR recordings appear at the topmost positions in the context menu. This has been requested by or users for several times.

The code was runtime-tested on Linux an macOS, latest master.

@Jalle19 for review?

### CODE REVIEW

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
@Jalle19 any comments? I know it's pain to read the large diff, but that's why i split it up into multiple commits. ;-) Stuff is well runtime tested for some time now, btw. Like always I use my family members as beta testers in the living room. No complaints so far. ;-)

**Comment by Jalle19** [comment]
Sorry, missed this one.

**Comment by ksooo** [comment]
rebased, good to go?

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/FileItem.cpp [hunk 1]
```diff
@@ -3533,17 +3534,42 @@ int CFileItem::GetVideoContentType() const
...
 
+CFileItem CFileItem::GetItemToPlay() const
+{
+  if (HasEPGInfoTag())
+  {
+    const CPVRChannelPtr channel(GetEPGInfoTag()->ChannelTag());
+    if (channel)
+      return CFileItem(channel);
+  }
+  return CFileItem(*this);
```
Comment (Jalle19) [inline xbmc/FileItem.cpp]: Wouldn't `return *this;` have the same effect as this but save one copy?

Comment (ksooo) [inline xbmc/FileItem.cpp]: Hehe, good catch.

Comment (Jalle19) [inline xbmc/FileItem.cpp]: Even better would be to potentially use `CFileItemPtr` instead, but that probably requires more careful thought.

Comment (ksooo) [inline xbmc/FileItem.cpp]: This would require to derive CFileItem from std::enable_shared_from_this which I wanted to prevent.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
Jenkins macOS error is not related to this PR.

<pre>
make: *** [/Users/Shared/jenkins/workspace/OSX-64/tools/depends/pre-depends/autoconf-pre-depends/../../pre-build-deps/autoconf-2.68.tar.gz] Error 7
autoconf was not found and could not be built. Aborting.
</pre>

**Pull request merged**